### PR TITLE
Print out the mining state

### PR DESCRIPTION
**Pull request by SWvheerden**
## Description

Prints out if your mining or not
Prints out the hash rate(estimated by last recorded thread hash rate * thread count)

## Motivation and Context

This will print out the cli the state of the mining. Currently mining state is only available if you read the log file or restart it. This will provide feedback to the user on the cli

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [ ] I'm merging against the `development` branch.
* [ ] I ran `cargo-fmt --all` before pushing.
* [ ] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: applications/tari_base_node/src/parser.rs [hunk 1]
```diff
@@ -375,6 +384,9 @@ impl Parser {
             ToggleMining => {
                 println!("Enable or disable the miner on this node, calling this command will toggle the state");
             },
+            GetMiningState => {
+                println!("This gets the mining state and prints out if its mining and the last known hash rate")
```
Comment (CjS77) [inline applications/tari_base_node/src/parser.rs]:
```suggestion
                println!("Displays the mining state. The hash rate is estimated based on the last measured hash rate and the number of active mining thread.")
```

File: base_layer/core/src/mining/blake_miner.rs [hunk 2]
```diff
@@ -65,6 +65,7 @@ impl CpuBlakePow {
                     std::u64::MAX - last_measured_nonce + nonce
                 };
                 let hash_rate = hashes as f64 / start.elapsed().as_micros() as f64;
+                hashrate.store((hash_rate * 1000.0) as u64, Ordering::Relaxed);
```
Comment (CjS77) [inline base_layer/core/src/mining/blake_miner.rs]: Why not * 1000000 and go straight to H? u64 is a huge number

**Review by CjS77 — COMMENTED** [review]
Looks good; Some small suggested tweaks is all.

File: base_layer/core/src/mining/blake_miner.rs (hunk 2 above)
Comment (SWvheerden) [reply base_layer/core/src/mining/blake_miner.rs]: fair enough.  updating

**Review by CjS77 — APPROVED** [review]
LGTM

**Pull request merged**